src/anomalib/models/efficientad/lightning_model.py

```diff
@@ -0,0 +1,248 @@
+"""EfficientAD: Accurate Visual Anomaly Detection at Millisecond-Level Latencies.
+https://arxiv.org/pdf/2303.14535.pdf
+"""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+
+import albumentations as A
+import numpy as np
+import torch
+import tqdm
+from albumentations.pytorch import ToTensorV2
+from omegaconf import DictConfig, ListConfig
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+from torch import Tensor, optim
+from torch.utils.data import DataLoader
+from torchvision.datasets import ImageFolder
+
+from anomalib.data.utils import DownloadInfo, download_and_extract
+from anomalib.models.components import AnomalyModule
+
+from .torch_model import EfficientADModel
+
+logger = logging.getLogger(__name__)
+
+IMAGENETTE_DOWNLOAD_INFO = DownloadInfo(
+    name="imagenette2.tgz",
+    url="https://s3.amazonaws.com/fast-ai-imageclas/imagenette2.tgz",
+    hash="fe2fc210e6bb7c5664d602c3cd71e612",
+)
+
+
+class TransformsWrapper:
+    def __init__(self, t: A.Compose):
+        self.transforms = t
+
+    def __call__(self, img, *args, **kwargs):
+        return self.transforms(image=np.array(img))
+
+
+class EfficientAD(AnomalyModule):
+    """PL Lightning Module for the EfficientAD algorithm."""
+
+    def __init__(
+        self,
+        category_name: str,
+        teacher_file_name: str,
+        teacher_out_channels: int,
+        pre_trained_dir: str,
+        image_size: list,
+        model_size: str = "M",
+        lr: float = 0.0001,
+        weight_decay: float = 0.00001,
+        padding: bool = False,
+    ) -> None:
+        super().__init__()
+
+        self.category = category_name
+        self.pre_trained_dir = Path(pre_trained_dir)
+        self.imagenet_dir = Path("./datasets/imagenette")
+        self.model: EfficientADModel = EfficientADModel(
+            teacher_path=self.pre_trained_dir / teacher_file_name,
+            teacher_out_channels=teacher_out_channels,
+            input_size=image_size,
+            model_size=model_size,
+            padding=padding,
+        )
+
+        self.data_transforms_imagenet = A.Compose(
+            [  # We obtain an image P ∈ R 3×256×256 from ImageNet by choosing a random image,
+                A.Resize(image_size[0] * 2, image_size[1] * 2),  # resizing it to 512 × 512,
+                A.ToGray(p=0.3),  # converting it to gray scale with a probability of 0.3
+                A.CenterCrop(image_size[0], image_size[1]),  # and cropping the center 256 × 256 pixels
+                A.ToFloat(always_apply=False, p=1.0, max_value=255),
+                ToTensorV2(),
+            ]
+        )
+
+        self.prepare_imagenet_data()
+        imagenet_dataset = ImageFolder(self.imagenet_dir, transform=TransformsWrapper(t=self.data_transforms_imagenet))
+        self.imagenet_loader = DataLoader(imagenet_dataset, batch_size=1, shuffle=True, pin_memory=True)
```

Comment:
would hard-coding the batch size here lead to problems when Anomalib's training batch size is set to >1?